File: Java/Matrizes/Matriz_geral.java
package Java.Matrizes;

import java.util.Locale;
import java.util.Scanner;



public class Matriz_geral {
    public static void main(String[] args) throws Exception {
        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);

        int M, i, j, linha, coluna;
        double somaPositivos;
        
        
        System.out.print("Qual a ordem da matriz? ");
        M = sc.nextInt();



        double [][] mat = new double[M][M];


        System.out.println("Digite os valores da matriz A:");

        for (i = 0; i < M; i++) {
            for (j = 0; j < M; j++) {
                System.out.print("Elemento [" + i + "," + j + "]: ");
                mat[i][j]= sc.nextDouble();
            }
        }
        somaPositivos = 0;

        for (i = 0; i < M; i++) {
            
            for (j = 0; j < M; j++) {
                if (mat[i][j] >= 0) {
                    somaPositivos = somaPositivos + mat[i][j];
                }
            }
        }

        System.out.println();
        System.out.println("SOMA DOS POSITIVOS = " + String.format("%.1f", somaPositivos));

        System.out.println();
        System.out.print("Escolha uma linha: ");
        linha = sc.nextInt();
        System.out.print("LINHA ESCOLHIDA:");

        for (i = 0; i < M; i++) {
            
            for (j = 0; j < M; j++) {
                if (linha == i){
                    System.out.print(mat[i][j] + "  ");
                }
            }
        }

        System.out.println();
        System.out.println();
        System.out.print("Escolha uma coluna: ");
        coluna = sc.nextInt();
        System.out.print("COLUNA ESCOLHIDA:");

        for (i = 0; i < M; i++) {
            
            for (j = 0; j < M; j++) {
                if (coluna == j){
                    System.out.print(mat[i][j] + "  ");
                }
            }
        }

        System.out.println();
        System.out.println();
        System.out.print("DIAGONAL PRINCIPAL: ");


        for (i = 0; i < M; i++) {
            
            for (j = 0; j < M; j++) {
                if (i == j){
                    System.out.print(mat[i][j] + "  ");
                }
            }
        }

        for (i = 0; i < M; i++) {
            
            for (j = 0; j < M; j++) {
                if (mat[i][j] < 0){
                    mat[i][j] = mat[i][j] * mat[i][j]; 
                }
            }
        }


        System.out.println();
        System.out.println();
        System.out.println("MATRIZ ALTERADA: ");


        for (i = 0; i < M; i++) {
            
            for (j = 0; j < M; j++) {
                System.out.print(mat[i][j] + "  ");
            }
            System.out.println();
        }
      


        sc.close();
    }
}

File: Python/Matrizes/Matriz_geral.py
N = int(input("Qual a ordem da matriz? "))

mat = [[0 for x in range(N)] for x in range(N)]

for i in range(0, N):
    for j in range(0, N):
        mat[i][j] = float(input(f"Elemento [{i},{j}]: "))

soma = 0

for i in range(0, N):
    for j in range(0, N):
        if mat[i][j] >= 0:
            soma = soma + mat[i][j]

print()
print(f"SOMA DOS POSITIVOS: {soma:.1f}")

print()
linha = int(input("Escolha uma linha: "))
print("LINHA ESCOLHIDA: ", end= " ")

for i in range(0, N):
    for j in range(0, N):
        if linha == i:
            print(f"{mat[i][j]:.1f}", end= " ")

print()
print()
coluna = int(input("Escolha uma coluna: "))
print("COLUNA ESCOLHIDA: ", end= " ")

for i in range(0, N):
    for j in range(0, N):
        if coluna == j:
            print(f"{mat[i][j]:.1f}", end= " ")


print()
print()
print("DIAGONAL PRINCIPAL: ", end= " ")

for i in range(0, N):
    for j in range(0, N):
        if i == j:
            print(f"{mat[i][j]:.1f}", end= " ")



for i in range(0, N):
    for j in range(0, N):
        if mat[i][j] < 0:
            mat[i][j] = mat[i][j] * mat[i][j]


print()
print()
print("MATRIZ ALTERADA:")

for i in range(0, N):
    for j in range(0, N):
        print(f"{mat[i][j]:.1f}", end=" ")
    print()